src/test/javascript/spec/component/course/course-group.component.spec.ts

```diff
@@ -75,130 +68,145 @@ describe('Course Management Detail Component', () => {
-            searchStub = sinon.stub(userService, 'search');
+            searchStub = jest.spyOn(userService, 'search').mockImplementation();
+        });
+
+        afterEach(() => {
+            jest.restoreAllMocks();
         });
 
         it('should search users for given login or name', () => {
-            searchStub.returns(of(new HttpResponse({ body: [courseGroupUser] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([courseGroupUser]);
+                expect(users).toEqual([courseGroupUser]);
             });
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
 
         it('should set search no results if search returns no result', () => {
-            searchStub.returns(of(new HttpResponse({ body: [] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(comp.searchNoResults).to.equal(true);
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(comp.searchNoResults).toBe(true);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
 
         it('should return empty if search text is shorter than three characters', () => {
             loginStream = of({ text: 'ab', entities: [] });
-            searchStub.returns(of(new HttpResponse({ body: [courseGroupUser] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(searchStub).not.to.have.been.called;
+            expect(searchStub).not.toHaveBeenCalled();
         });
 
         it('should return empty if search fails', () => {
-            searchStub.returns(throwError(new Error('')));
+            searchStub.mockReturnValue(throwError(new Error('')));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([]);
+                expect(users).toEqual([]);
             });
-            expect(comp.searchFailed).to.be;
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(comp.searchFailed).toBe(true);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
         });
     });
 
     describe('onAutocompleteSelect', () => {
-        let addUserStub: sinon.SinonStub;
-        let fake: sinon.SinonSpy;
+        let addUserStub: jest.SpyInstance;
+        let fake: jest.Mock;
```

Comment:
I don't see the sense behind defining it in this scope. I suggest moving it into the tests and just passing `jest.fn()`.
If you want to keep it, you have to manually reset it.

Reply:
I refactored it. I talked with @MDK5 about this and we found out that it makes no difference if it does not get reset.